### DISCUSSION TITLE
Future of Isar/Hive

### DISCUSSION THREAD

**Discussion by ghost** [isar/isar#1498] [Help]
I'm a bit worried because Isar and Hive seem to be a bit abandoned by Simon. It's been 3 months since the last commit and Simon isn't replying on Telegram either? Does anyone have more information?

**Comment by mafanwei** [comment]
I share your concerns.

**Reply by abdullah-cse** [reply]
Do you have personal connection with @simc ?

**Comment by aidevani** [comment]
That would be sad. I tried Hive and moved to Isar as my core use-case involves multi-isolate usage. And Isar seems great at that. 
Are there any other databases that support multi-isolate usage in flutter?

**Comment by ghost** [comment]
@Jtplouffe, as you are the only other member of isar besides @simc. Do you have more information?

**Comment by morhi** [comment]
I share your concerns too. It actually brings me into some trouble because a) I didn't find another solution (Offline, NoSQL, Schemas) yet, that supports web too b) it will be difficult to migrate all data from the phones to another database. I love Isar, so I'd prefer to continue to use it, but the history of Isar suggests, that it is not the best idea in the long-term.

**Reply by ghost** [reply]
I have exactly the same problem, Isar/Hive was a complete game changer in terms of web support. But I also think that Isar needs to be maintained by the community and not just one person to be a reliable solution in the long run.

**Comment by abdullah-cse** [comment]
This discussion should be closed now. @simc is back on Telegram. He was busy with personal life. All are OK! Use Isar and Hive confidently

**Reply by morhi** [reply]
@abdullah-cse As @juliaen-dr suggested, this does not remove the need for a solution. This has happened in the past already and makes this awesome project unreliable, where reliability matters a lot for an important package like a database in a project. You can't just swap this package for another one.

**Comment by ghost** [comment]
I don't think we should close this now. I think there needs to be a solution on how we can reduce workload for @simc and spread responsibilities for the whole project.

**Reply by morhi** [reply]
@juliaen-dr I remember that there was a discussion on this in the past already, but I am not sure if it was a GitHub issue or something else. Do you remember?

**Reply by ghost** [reply]
Sadly not, maybe this was before I started using Hive/Isar

**Reply by ghost** [reply]
Maybe this one https://github.com/isar/isar/discussions/1153 ?

**Comment by mafanwei** [comment]
What can I do? Do you have a todo list?

**Comment by mafanwei** [comment]
I am happy to offer my help to this open-source project.

**Comment by ghost** [comment]
I think the next step would be to wait for @simc's opinion.

**Reply by mafanwei** [reply]
But who has personal connection with @simc ?

**Reply by mafanwei** [reply]
I'm worried that he won't be able to see

**Reply by ghost** [reply]
He answered on Telegram that he will be back soon. I'm sure he will see this thread

**Reply by mafanwei** [reply]
Great !

**Reply by deployHuman** [reply]
And yet time passes on,, i bet there would be some great people here that can help.
Cant we put up a sort of voting for who we like to be a part of the admins of the repo ? so more ppl can merge new fixes etc

**Reply by ghost** [reply]
As long as @simc is not answering to this post, I think there is nothing we can do

**Comment by alexobviously** [comment]
Gotta say that it is somewhat concerning that the health of the most used local storage package for dart depends entirely on the availability of one person, and we should really find a solution for this sooner rather than later.

Also worth thinking about in the context of all the other important packages that depend heavily on one guy (e.g. bloc).

**Reply by huntrist** [reply]
is it possible for flutter community to maintain a fork (or maybe make this project part of the flutter community)?
make it somewhat like "share_plus", "connectivity_plus", etc. (i.e. packages here: https://pub.dev/publishers/fluttercommunity.dev/packages)

don't know if then will have more people helping maintain it.

**Comment by zhanchen18** [comment]
Yet a month has passed since November, and still nothing has happened

**Comment by ghost** [comment]
To be honest I don't think this project will reach some reliable stage in the near future

**Comment by spatialbits** [comment]
He's back :)

**Comment by Dumoulin-Lander** [comment]
@simc I imagine reading issues like these might rub you the wrong way, as people that are using your (great) solution for free may seem entitled when having complaints. Especially when you have personal things to deal with that are more important to you than this project (as they should be!)

However, I think it's in the best interest of both the project and your sanity that you create a path forward where the project can somewhat move forward without relying so heavily on you as its main development resource. 

Can you conceive of a good way to incorporate more community members' contributions? Can you delegate tasks in the short term?

**Reply by alexobviously** [reply]
This is a very good comment and totally correct. I think we all totally support @simc and wish him the best with whatever he's dealing with, and probably quite a lot of us are also open source maintainers and understand the struggle well. None of this is a criticism of you at all, it's just something we need to solve, and part of a wider problem that we as the open source community need to solve together. Thanks for all of your work!

**Comment by tschiekdev** [comment]
We are back to the point where it pisses me off that the Google Flutter developers do not create their own internal NoSQL database. I find data storage to be something fundamental and in my opinion that should be included. And no, I don't mean firebase. :-(

**Comment by rivella50** [comment]
Since @simc doesn't seem to resume his work on Isar and Hive in the near future, could it be an option that some enthusiasts of these 2 packages continue his work under new package names, and if @simc returns at some point of time the packages could be merged?
Imo those 2 databases are too important for just let them degenerate.

**Comment by alexobviously** [comment]
I think a fork is a sensible course of action at this point. But having said that, maybe a total replacement is a possibility. I just worry that Hive is so far behind the latest Dart features, that dealing with the technical debt might be more work than planning and building something from scratch. Another concern for me as a potential contributor to a fork is that Hive now seems to rely on Isar, which I have no idea about, and which seemingly has its own set of tech debt (doesn't work on web, huge regression from pre-Isar-Hive!) Of course we'd have Hive for reference, but idk, just putting it out there that it might be the right course of action to build something new.

**Reply by rivella50** [reply]
The Hive prerelease version 4.0.0-dev.2 isn't that old yet and probably already contains some good aspects for a new version.

**Reply by alexobviously** [reply]
How stable is it? I haven't used it at all because of the lack of web support, so I don't have much of an idea - every project I have with Hive is on 2.x.x.

**Reply by rivella50** [reply]
I've used it in some demo projects and it works well already (never crashed so far). But since it's a dev version i (of course) wouldn't use it for a released app.

**Comment by MrCsabaToth** [comment]
Will there ever be Isar v4? It's been 5-6 months since the last dev version.

**Comment by ghost** [comment]
Would be nice if @Jtplouffe let's us know about his thoughts since he is the only one besides Simon working on these projects

**Comment by fedikhatib** [comment]
Today simon was on podcast about isar 
https://youtu.be/InTuwdeTNJ0

**Reply by AristideVB** [reply]
@fedikhatib This interview is 6 month old ! ⚠️⚠️🥲

<img width="1048" alt="Capture d’écran 2024-02-16 à 12 25 50" src="https://github.com/isar/isar/assets/97469482/6a4549c6-3e03-4934-94c4-d77b960b3bb6">

![IMG_D3463A3D87BE-1](https://github.com/isar/isar/assets/97469482/d689bdbf-6eff-4999-a17e-4174ef1ad86d)

**Reply by abdullah-cse** [reply]
Pre recorded video of 11 august 2023. See the right+top corner for date

**Reply by outdoorapps** [reply]
I was waiting for the Isar v4 update. Glad that I read this thread just now. Looks like the project wouldn't be resumed in the near future. I am now switching to objectbox for sure. No offense, I love Isar for its simplicity and performance but it is hard to rely on one person to develop a package for my project. I would still love to see this project come back to life in the future.

**Comment by mdmm13** [comment]
FYI there's a discussion on Telegram around creating a community fork to close open issues and at least keep it compatible with new Flutter/ Dart versions until Simon returns. Anyone interested, please take a look. Proposed repo is https://github.com/isar-community.

**Reply by rivella50** [reply]
Cool, is something similar also planned for Hive?

**Reply by mdmm13** [reply]
Hive v4 is just a wrapper on top of Isar v4. 

If someone wants to create a Hive-Community wrapper and rebase it onto the Isar-Community fork, that's totally fine. We're still trying to find people who can actually help drive the Isar-Community fork.

**Comment by JulienElkaim** [comment]
Did this conversation moved anywhere else? 

After a quick benchmark of Flutter NoSQL alternatives, Isar is still a great choice. 
However, if not maintained, I am afraid to rely on it for my projects.

@simc  please let us know if you need help on Isar, if you are on a personal project and need to discontinue Isar/Hive for good, or anything else.
 
I have the feeling by reading this conversation that many people would be happy to help you.

**Reply by AristideVB** [reply]
@juliaen-dr for anyone wanting to contribute to the future of Isar you can here https://github.com/isar-community ! 🙂
